app/controllers/searches_controller.rb

```diff
@@ -1,8 +1,10 @@
 class SearchesController < ApplicationController
   before_action :set_page, only: :show
+  LAST_PAGE = 333 # In ES, query offset can't go beyond 10_000 records
 
   def show
     return unless params[:query] && params[:query].is_a?(String)
+    @page = LAST_PAGE if @page > LAST_PAGE
```

Comment:
I know this is really edge case, but what about to show proper warning or error message?